docs/vm-ee-guide.md

```diff
@@ -0,0 +1,324 @@
+# Onboard
+
+## Table of Contents
+
+<!-- toc -->
+- [VirtualMachines](#virtualmachines)
+    - [Labels](#labels)
+    - [Cloud Tags](#cloud-tags)
+    - [Sample VirtualMachine Output](#sample-virtualmachine-output)
+        - [AWS](#aws)
+        - [Azure](#azure)
+- [ExternalEntity](#externalentity)
+    - [Sample ExternalEntity Output](#sample-externalentity-output)
+        - [AWS](#aws-1)
+        - [Azure](#azure-1)
+    - [Labels](#labels-1)
+    - [Cloud Tags](#cloud-tags-1)
+        - [Usage](#usage)
+<!-- /toc -->
+
+## Virtual Machine
+
+Virtual Machines are onboarded by specifying the matching criteria in `CloudEntitySelector` CR. The virtual machines from
+cloud are imported as `VirtualMachine` objects in `CloudEntitySelector` namespace. `VirtualMachine` object `Name` is
+unique resource name within the kubernetes cluster. In Azure, ascii sum of the characters in VM Resource ID added as
+suffix to name of the Virtual Machine and in AWS, instance ID of the Virtual Machine is used as `Name`.
+
+### Labels
+
+Meaning of some label fields in `VirtualMachine` objects differ based on the cloud provider type.
+
+| Label        | Description                               | Azure                                                   | AWS                        |
+|--------------|-------------------------------------------|---------------------------------------------------------|----------------------------|
+| cloud-vm-uid | Unique identifier of VM on cloud          | VM `vmId` property on cloud                             | Instance ID of VM on cloud |
+| cloud-vpc-id | Unique identifier of VPC on cloud         | VNET `resourceGuid` property on cloud                   | VPC ID of VPC on cloud     |
+| vpc-name     | VPC object name within Kubernetes cluster | VNET `name` suffixed with ascii sum of VNET Resource ID | VPC ID of VPC on cloud     |
+
+Following labels are generic and related to user configuration.
+
+1. `cpa-name`: Name of `CloudProviderAccount` CR to which the imported VM belongs to.
+2. `cpa-namespace`: Namespace of `CloudProviderAccount` CR to which the imported VM belongs to.
+
+### Cloud Tags
+
+`VirtualMachine` object tags are converted to labels in `ExternalEntity` CR, hence only the tags which comply to
+Kubernetes labels format are imported as tags in VM object.
```

Comment:
The tags of `VirtualMachine`
CR. Therefore, only tags that comply to the Kubernetes label format are imported in the VM object.

Reply:
done